java/code/src/com/redhat/rhn/manager/access/AccessGroupManager.java

```diff
@@ -0,0 +1,195 @@
+     * @param org the org to which the group belongs
+     * @return the created {@code AccessGroup}
+     */
+    public AccessGroup create(String label, String description, Org org) throws DefaultRoleException {
+        return this.create(label, description, org, Collections.emptyList());
+    }
+
+    /**
+     * Creates a new access group with the given parameters.
+     * @param label the label of the group
+     * @param description the description of the group
+     * @param org the org to which the group belongs
+     * @param inheritFrom a collection of {@code AccessGroup}s to inherit permissions from
+     * @return the created {@code AccessGroup}
+     */
+    public AccessGroup create(String label, String description, Org org, Collection<AccessGroup> inheritFrom)
+            throws DefaultRoleException {
+        if (ORG_ADMIN.getLabel().equals(label) || SAT_ADMIN.getLabel().equals(label) ||
+                lookup(label, null).isPresent()) {
+            throw new DefaultRoleException(label + " already exists.");
+        }
+        AccessGroup group = new AccessGroup(label, description, org);
+        for (AccessGroup parent : inheritFrom) {
+            addPermissionsFromRole(group, parent);
+        }
+
+        AccessGroupFactory.save(group);
+        return group;
+    }
+
+    /**
+     * Removes an access group with the given label and org.
+     * @param label the label of the access group to remove
+     * @param org the org to which the access group belongs
+     * @throws java.util.NoSuchElementException if the {@code AccessGroup} is not found.
+     */
+    public void remove(String label, Org org) {
+        AccessGroup group = lookup(label, org).orElseThrow();
+        if (group.getOrg() == null) {
+            throw new IllegalArgumentException("Default groups cannot be altered.");
+        }
+        AccessGroupFactory.remove(group);
+    }
+
+    /**
+     * Lists all access groups defined in MLM.
+     * @return a list of access groups
+     */
+    public List<AccessGroup> list() {
+        return AccessGroupFactory.listAll();
+    }
+
+    /**
+     * Lists all access groups belonging to the given org.
+     * @param org the org to list access groups from
+     * @return a list of access groups
+     */
+    public List<AccessGroup> list(Org org) {
+        return AccessGroupFactory.list(org);
+    }
+
+    /**
+     * Looks up an access group with the given label and org
+     * @param label the label of the group to look up
+     * @param org the org to which the group belongs
+     * @return an {@code Optional} containing the access group, or an empty {@code Optional} if not found
+     */
+    public Optional<AccessGroup> lookup(String label, Org org) {
+        return AccessGroupFactory.lookupByLabelAndOrg(label, org);
+    }
+
+    /**
+     * Grants access to the given namespaces for the specified access group.
+     * <p>
+     * The asterisk (*) character can be used as a wildcard in namespace strings.
```

Comment:
I think someone in the presentaton asked about the need for the `*` and if we should just remove it and state that anything in that namespace and "children" will be added. The reality is that in the query processing you are just replacing the `*` with `%`. By not having it in the input you could just concatenate the `%` at the end.

Reply:
The reason I didn't want to do it is this case:

Imagine there are 3 namespaces like following:
 - foo
 - foo.baz
 - foobar.baz
 
 If `%` is added by default, a query with `foo` will return `foo`, `foo.baz`, `foobar.baz`.
 
 By making it explicit, the user will have control over it:
 
  - `foo` would return only `foo`
  - `foo.*` would return `foo.baz`
  - `foo*` would return `foo`, `foo.baz` and `foobar.baz`
  
  I made sure that this problem doesn't appear with the initial data, but since we'll be adding namespaces constantly, it's not easy to prevent this in the future, so I think making `*` explicit is a safer way.

Reply:
Makes sense to me. If users don't like it or don't understand it we can always revisit it.